tests/specs/lemmas.k

```diff
@@ -241,6 +241,9 @@ module LEMMAS-HASKELL [symbolic, kore]
     imports EVM
     imports EDSL
 
+    rule #lookup ( M:Map [ K1 <- V1 ] , K2 ) => #lookup ( K1 |-> V1 , K1 ) requires K1 ==Int  K2 [simplification]
+    rule #lookup ( M:Map [ K1 <- _  ] , K2 ) => #lookup ( M         , K2 ) requires K1 =/=Int K2 [simplification]
+
```

Comment:
Can these be added to just `LEMMAS` instead to be used on both Haskell and Java? Or is it only needed on Haskell?

Reply:
let me try

Reply:
done. see if tests pass